### PR TITLE
Algolia search bar config - merge March 15

### PR DESCRIPTION
**Pull request by kimsauce**
## Purpose of this pull request

This pull request adds Algolia and removes ss360.

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions and updating sections
- [ ] New Content - New features, sections, pages, tutorials
- [x] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

### CODE REVIEW

**Review by stacykor — APPROVED** [review]
LGTM

File: docusaurus.config.js [hunk 1]
```diff
@@ -383,9 +379,9 @@ module.exports = {
             value: '<div id="google_translate_element"></div>',
           },
           {
-            to: 'https://www.sumologic.com/sign-up',
-            position: 'right',
             className: 'header-login',
+            position: 'right',
+            to: 'https://www.sumologic.com/sign-up',
```
Comment (kimsauce) [inline docusaurus.config.js:384]:
```suggestion
            to: 'https://www.sumologic.com/sign-up',
            position: 'right',
```

File: docusaurus.config.js [hunk 2]
```diff
@@ -383,9 +379,9 @@ module.exports = {
             value: '<div id="google_translate_element"></div>',
           },
           {
+            className: 'header-login',
```
Comment (kimsauce) [inline docusaurus.config.js:382]:
```suggestion
```

File: src/theme/SearchBar/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import SearchBar from '@theme-original/SearchBar';
+
+export default function SearchBarWrapper(props) {
+  return (
+    <>
```
Comment (irontitan76) [inline src/theme/SearchBar/index.js:6]: This fragment isn't necessary.

**Comment by irontitan76** [comment]
Are you trying to get this in between the `Start a Free Trial` button and the other icon buttons?

File: src/theme/SearchBar/index.js (hunk 3 above)
Comment (kimsauce) [reply src/theme/SearchBar/index.js:6]: Removing this section (and the file, too) breaks the build. Can you advise?

**Comment by kimsauce** [comment]
> Are you trying to get this in between the `Start a Free Trial` button and the other icon buttons?

Right now I have it in the far right. Open to moving it elsewhere in the nav, though.

File: docusaurus.config.js [hunk 4]
```diff
@@ -386,6 +382,7 @@ module.exports = {
             to: 'https://www.sumologic.com/sign-up',
             position: 'right',
             className: 'header-login',
+            'aria-label': 'Start a free trial',
```
Comment (kimsauce) [inline docusaurus.config.js:385]:
```suggestion
```

File: docusaurus.config.js [hunk 5]
```diff
@@ -330,6 +323,14 @@ module.exports = {
...
                 activeBaseRegex: '^/docs/(sdo)/.*',
               },
+              {
+                label: 'Dashboards',
+                to: '/docs/dashboards-new',
+              },
+              {
+                label: 'Contribute',
+                to: '/docs/contributing',
+              },
```
Comment (kimsauce) [inline docusaurus.config.js:333]:
```suggestion
```

**Comment by asachs01** [comment]
Hey folks, I've been working with @kimsauce to try and narrow down why the search results aren't being segmented out. The TL;DR is that I believe the terms used for faceting in Algolia (see [their guide on faceting here](https://www.algolia.com/doc/guides/solutions/ecommerce/business-users/initial-configuration/faceting/) need some tweaking in order to allow for a more robust search.

If we take the Sensu docs, for example, the results are segmented by the product: 
CleanShot 2023-01-31 at 10 53 00

In order to accomplish this, [there's some magic in Hugo](https://github.com/sensu/sensu-docs/blob/main/layouts/partials/head.html) in our headers partial that sets the `product` tag based on the front matter for the doc (see [this example guide](https://raw.githubusercontent.com/sensu/sensu-docs/main/content/sensu-go/6.9/catalog/build-private-catalog.md)). 

But for that to be usable by Algolia, we've configured the following facets in our index:
CleanShot 2023-01-31 at 09 53 00

Since that's set on the Algolia side, we've got our results narrowed down by the facetfilters we've added to our [footer partial here](https://github.com/sensu/sensu-docs/blob/main/layouts/partials/footer_js.html#L183-L217):

```
 algoliaOptions: {
      facetFilters: [
        [
          {{ if isset .Page.Params "product" }}
            'tags:{{ .Section }}'
          {{ else }}
            'tags:sensu-go',
            'tags:plugins',
            'tags:sensu-core',
            'tags:sensu-enterprise',
            'tags:sensu-enterprise-dashboard',
            'tags:uchiwa'
          {{end}}
        ],
        [
          {{ if isset .Page.Params "product" }}
            'version:{{ .Page.Params.version }}'
          {{ else }}
            'version:6.9',
            'version:1.0',
            'version:2.16',
            'version:1.9',
            'version:3.7'
          {{end}}
        ]
      ]
    },
    debug: false
  };
```

But from what I'm seeing in the Sumo Algolia config, it doesn't look like the facets are set correctly:

CleanShot 2023-01-31 at 09 56 35

And it doesn't look like facetfilters are being used as part of the Algolia configuration (see https://github.com/SumoLogic/sumologic-documentation/blob/615bd933566e118f3d8e1b309761941559d11f13/docusaurus.config.js#L241-L253).

That said, I think if we could narrow down the facets that we want for grouping (I'm thinking that the product would be one of the facets that we group the results by) and then get that configured in the docusaurus config, that should get us the desired results.

**Comment by asachs01** [comment]
Wahey!!! It looks like it's working @kimsauce image

**Comment by kimsauce** [comment]
@asachs01 yes!! Oh my god, after a month of trial and error, it finally worked! It was your guidance about the faceting that got me almost to the finish line, then after a couple days of tweaking the dashboard config, I got it😊😄 So so happy.

Also thank you @vfalconisumo for your work figuring out the "See all results" link! I'm still trying to learn how to build out the "/search" landing page.

File: src/components/NavLink/styles.css [hunk 6]
```diff
@@ -3,7 +3,7 @@
 }
 
 .link__label {
-    font-size: 0.75rem !important;
+    font-size: 0.8rem !important;
```
Comment (kimsauce) [inline src/components/NavLink/styles.css:6]:
```suggestion
    font-size: 0.75rem !important;
```

File: src/theme/SearchBar/index.js (hunk 3 above)
Comment (kimsauce) [reply src/theme/SearchBar/index.js:6]: Hi @irontitan76 , to clarify, were you advising that we remove the whole SearchBar component (src/theme/SearchBar/index.js)? I've tried doing this, but without it, the Algolia search bar won't appear in the nav.

File: docs/send-data/use-json-configure-sources/index.md [hunk 7]
```diff
@@ -2,6 +2,7 @@
 slug: /send-data/use-json-configure-sources
 title: Use JSON to Configure Sources
 description: Learn how to configure sources and processing rules using JSON.
+tags: [json]
```
Comment (kimsauce) [inline docs/send-data/use-json-configure-sources/index.md:5]:
```suggestion
```

File: docs/observability/kubernetes/index.md [hunk 8]
```diff
@@ -2,6 +2,7 @@
 slug: /observability/kubernetes
 title: Kubernetes Solution
 description: This guide provides and end-to-end solution for deploying, managing, monitoring, and administering your Kubernetes environment.
+tags: [kubernetes]
```
Comment (kimsauce) [inline docs/observability/kubernetes/index.md:5]:
```suggestion
```

**Pull request merged**